flytekit/extend/backend/base_plugin.py

```diff
@@ -0,0 +1,66 @@
+import typing
+from abc import ABC, abstractmethod
+
+import grpc
+from flyteidl.core.tasks_pb2 import TaskTemplate
+from flyteidl.service.external_plugin_service_pb2 import (
+    RETRYABLE_FAILURE,
+    RUNNING,
+    SUCCEEDED,
+    State,
+    TaskCreateResponse,
+    TaskDeleteResponse,
+    TaskGetResponse,
+)
+
+from flytekit.models.literals import LiteralMap
+
+
+class BackendPluginBase(ABC):
```

Comment:
If this is the main Base class that people will be implementing, can we add a lot more comments here?